packages/uniforms/src/SimpleSchemaBridge.js

```diff
@@ -202,8 +202,8 @@ export default class SimpleSchemaBridge extends Bridge {
             } else {
                 field = {
                     ...field,
-                    transform: value => options.find(option => option.value === value).label,
-                    allowedValues: options.map(option => option.value)
+                    transform: value => options.find(option => option.value.toString() === value).label,
+                    allowedValues: options.map(option => option.value.toString())
```

Comment:
Hi @radekmie , I don't how that should work.  The problem is in the transform function. 

Just for being sure, this is what you suggest?

```html
    <Select
        allowClear={!props.required}
        disabled={props.disabled}
        id={props.id}
        multiple={props.fieldType === Array}
        name={props.name}
        onChange={value => props.onChange(value)}
        ref={props.inputRef}
        value={props.fieldType === Array
            ? props.value.map(item => item.toString) || []
            : props.value
                ? props.transform ? '' + props.transform(props.value.toString()) : '' + props.value
                : ''}
        {...filterDOMProps(props)}
    >

        {(!!props.placeholder || !props.required) && (
            <Select.Option value="" disabled={props.required} hidden={props.required}>
                {props.placeholder ? props.placeholder : props.label}
            </Select.Option>
        )}

        {props.allowedValues.map(value =>
            <Select.Option key={value} value={value}>
                {props.transform ? props.transform(value.toString()) : value}
            </Select.Option>
        )}
    </Select>
```

The real problem is in the `transform` function. The `value` is allways a string
The problem is that the transform function will be comparing the `value` string to the set of allowed values that might be numbers. So they will never be equal.